core/server/data/migrations/versions/4.3/08-migrate-members-signup-setting.js

```diff
@@ -0,0 +1,96 @@
+const logging = require('../../../../../shared/logging');
+const ObjectId = require('bson-objectid');
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up(options) {
+        const oldSetting = await options.transacting('settings')
+            .where('key', 'members_allow_free_signup')
+            .select('value', 'created_by', 'updated_by')
+            .first();
+
+        // no need to create new setting in this case, Ghost will create through default settings
+        if (!oldSetting) {
+            logging.warn('Could not find setting `members_allow_free_signup`, skipping update of `members_signup_access` setting');
+            return;
+        }
+
+        const newSetting = await options.transacting('settings')
+            .where('key', 'members_signup_access')
+            .select('value')
+            .first();
+
+        const migrateValue = oldSetting.value ? 'all' : 'invite';
+
+        if (newSetting) {
+            // new setting already exists, *update* with migrated value
```

Comment:
I was tripped up by this a few times whilst testing in development. Normally this state wouldn't be reachable but until 4.3 is released we can end up in a half-way state:

1. Local instance updated to `main` with new migration
2. Ghost started, warning shown `WARN Please run `knex-migrator migrate --v 4.3 --force` to proceed!` but boot continues and creates default settings
3. `knex-migrator migrate --v 4.3 --force` is run

Reply:
Gross! Good catch though :grin: